Source/Core/Core/ReliableUDPManager/ReliableUDPManager.cpp

```diff
@@ -0,0 +1,317 @@
+	for (int i = 0; i < 100; ++i)
+	{
+		sf::Socket::Status stat = m_udpSocket->receive(pack, ip, port);
+
+		if (stat == sf::Socket::Done)
+		{
+			//IpAndPort inp(ip.toInteger(), port);
+			std::pair<int, unsigned short> inp = std::make_pair(ip.toInteger(), port);
+			auto itr = m_AddresstoConnection.find( inp );
+			if (m_listen && itr == m_AddresstoConnection.end())
+			{
+				//This is a new connection store it and start accepting packets
+				//Also store it as a new connection ID so the user can decide 
+				//if he wants it
+				std::shared_ptr<ReliableUDPConnection> newConn = std::make_shared<ReliableUDPConnection>(m_udpSocket, ip, port);
+				int id = StoreConnection(newConn);
+				m_newConnection.push(id);
+				newConn->Receive(pack);
+				newConn->Send();
+			}
+			else if (itr != m_AddresstoConnection.end())
+			{
+				itr->second->Receive(pack);
+			}
+
+		}
+	}
+	
+	//--for now just send all messages in the message buffer
+	//--in the future we should just send a certain amount a bytes per 30 or 20 millisecond
+	std::list<u16> eraseList;
+	for (auto c : m_IDtoConnection)
+	{
+		if (!c.second->CheckIfAlive())
+		{
+			//if we havn't recieved a message for a long time, get rid of the connection
+			eraseList.push_front(c.first);
+			m_disconnection.push(c.first);
+			continue;
+		}
+	
+		do
+		{
+			c.second->Send();
+		} while (c.second->isReadyToSend());
+		
+	}
+
+	for (auto er : eraseList)
+	{
+		RemoveConnectionStore(er);
+	}
+
+}
+
+bool ReliableUDPManager::SendMess(u16 ID, sf::Packet& packet)
+{
+	auto con = m_IDtoConnection.find(ID);
+	if (con != m_IDtoConnection.end())
+	{
+		con->second->StoreSend(packet);
+		return true;
+	}
+	return false;
+}
+
+void ReliableUDPManager::SendMessageToAll(sf::Packet& packet, u16 exceptionID)
+{
+	for (auto c : m_IDtoConnection)
+	{
+		if (c.first != exceptionID)
+		{
+			c.second->StoreSend(packet);
+		}
+	}
+	
+}
+
+bool ReliableUDPManager::GrabMessage(u16 ID, sf::Packet& packet)
+{
+	auto con = m_IDtoConnection.find(ID);
+
+	if (con != m_IDtoConnection.end())
+	{
+		
+		return con->second->GrabMessage(packet);;
+	}
+	return false;
+}
+
+u16 ReliableUDPManager::CheckNewConnections(sf::IpAddress& ip, unsigned short& port)
+{
+	if (!m_newConnection.empty())
+	{
+		u16 id = m_newConnection.front();
+		m_newConnection.pop();
+
+		ip = m_IDtoConnection[id]->GetAddress();
+		port = m_IDtoConnection[id]->GetPort();
+
+		return id;
+	}
+	return 0;
+}
+
+
+u16 ReliableUDPManager::Connect(std::string ip, u16 port, float wait)
+{
+	bool oldListen = m_listen;
+	m_listen = true;
+	sf::Packet pack;
+	ReliableUDPConnection newConn(m_udpSocket, ip, port);
+	newConn.Send();
+	
+	Common::Timer sendTimer;
+	sendTimer.Start();
+	
+	Common::Timer giveUpTimer;
+	giveUpTimer.Start();
+
+	//keep trying to get a connection until our wait time is over
+	do
+	{
+		
+		Update();
+
+		//check new connection list to see if ours was added
+		sf::IpAddress ip;
```

Comment:
You may wish to change the name of this or the input parameter, as this will cause a variable shadowing warning.